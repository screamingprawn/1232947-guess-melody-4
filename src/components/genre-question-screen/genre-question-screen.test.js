import React from 'react';
import GenreQuestionScreen from './genre-question-screen.jsx';
import renderer from 'react-test-renderer';

const question = {
  type: `genre`,
  genre: `rock`,
  variants: [{
    src: `https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg`,
    genre: `rock`,
  }, {
    src: `https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg`,
    genre: `blues`,
  }, {
    src: `https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg`,
    genre: `jazz`,
  }, {
    src: `https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg`,
    genre: `rock`,
  }],
};

it(`GenreQuestionScreen is rendered correctly`, () => {
  const tree = renderer.create((
    <GenreQuestionScreen
      question={question}
      onAnswer={() => {}}
    />
  ), {
    createNodeMock: () => {
      return {};
    }
  }).toJSON();

  expect(tree).toMatchSnapshot();
});
