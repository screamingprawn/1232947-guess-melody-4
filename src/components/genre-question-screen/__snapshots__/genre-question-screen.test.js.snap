// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenreQuestionScreen is rendered correctly 1`] = `
<section
  className="game__screen"
>
  <h2
    className="game__title"
  >
    Выберите 
    rock
     треки
  </h2>
  <form
    className="game__tracks"
    onSubmit={[Function]}
  >
    <div
      className="track"
    >
      <button
        className="track__button track__button--play"
        type="button"
      />
      <div
        className="track__status"
      >
        <audio
          src="https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg"
        />
      </div>
      <div
        className="game__answer"
      >
        <input
          checked={false}
          className="game__input visually-hidden"
          id="variant-0"
          name="userAnswer"
          onChange={[Function]}
          type="checkbox"
          value="variant-0"
        />
        <label
          className="game__check"
          htmlFor="variant-0"
        >
          Отметить
        </label>
      </div>
    </div>
    <div
      className="track"
    >
      <button
        className="track__button track__button--play"
        type="button"
      />
      <div
        className="track__status"
      >
        <audio
          src="https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg"
        />
      </div>
      <div
        className="game__answer"
      >
        <input
          checked={false}
          className="game__input visually-hidden"
          id="variant-1"
          name="userAnswer"
          onChange={[Function]}
          type="checkbox"
          value="variant-1"
        />
        <label
          className="game__check"
          htmlFor="variant-1"
        >
          Отметить
        </label>
      </div>
    </div>
    <div
      className="track"
    >
      <button
        className="track__button track__button--play"
        type="button"
      />
      <div
        className="track__status"
      >
        <audio
          src="https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg"
        />
      </div>
      <div
        className="game__answer"
      >
        <input
          checked={false}
          className="game__input visually-hidden"
          id="variant-2"
          name="userAnswer"
          onChange={[Function]}
          type="checkbox"
          value="variant-2"
        />
        <label
          className="game__check"
          htmlFor="variant-2"
        >
          Отметить
        </label>
      </div>
    </div>
    <div
      className="track"
    >
      <button
        className="track__button track__button--play"
        type="button"
      />
      <div
        className="track__status"
      >
        <audio
          src="https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg"
        />
      </div>
      <div
        className="game__answer"
      >
        <input
          checked={false}
          className="game__input visually-hidden"
          id="variant-3"
          name="userAnswer"
          onChange={[Function]}
          type="checkbox"
          value="variant-3"
        />
        <label
          className="game__check"
          htmlFor="variant-3"
        >
          Отметить
        </label>
      </div>
    </div>
    <button
      className="game__submit button"
      type="submit"
    >
      Ответить
    </button>
  </form>
</section>
`;
